.github/workflows/checks.yml

```diff
@@ -42,6 +42,23 @@ jobs:
         run: |
           black --check .
 
+  pylint:
+    name: Code linting (pylint)
+    runs-on: ubuntu-latest
+    needs: [code-formatting]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+      - name: Install dev dependencies
+        run: |
+          pip install --progress-bar off -r requirements-dev.txt
+          pip list
+      - name: Run pylint
+        run: |
+          pylint watertap
```

Comment:
For the first implementation of this, can you just run pylint on a sub-directory of watertap? For example, do something like `pylint watertap/core` instead of linting the whole package. When linting passes in watertap/core then add another sub-directory and so on. Eventually, you can change this back to `pylint watertap` when all the sub-directories have passed.